### ISSUE TITLE
Excel text formatting information inconsistently read

### ISSUE THREAD

**Issue by Qchristensen** [nvaccess/nvda#6478]
Partly related to #3583 but also a bug in how formatting is currently read.

In an Excel worksheet (testing in Office 2007 and 2016), if text is typed into one cell with different formatting, eg:

Text can be **bold** or _italic_

(where the word Bold is actually bolded and the word italic is in italics)

When the focus moves to that cell, NVDA reads "No bold, no italic" before the contents (bold and italic in this case because those are the two properties that part of the text has).

If the first word was bold, NVDA would read "Bold, no italics".

Related: Aside from the round about way of knowing from that announcement that there is bold and italic in this cell, there is no way of finding out which text is formatted differently. I don't mind so much not having it directly read out as you navigate around the spreadsheet, but I would argue it should be identified as you move through and edit the text in that cell in edit mode (eg with the arrow keys after pressing F2).

NVDA reads the information about the first character, so if the whole cell is bold or underlined etc, that reads fine when navigating the workbook.

**Comment by jcsteh** [comment]
Reading changes to formatting in a cell while navigating a spreadsheet would be quite slow and arguably too verbose. However, instead of saying, for example, "no bold", we should at least be saying "bold" in the presence of mixed bold.

P2 because this is incorrect information.

**Comment by michaelDCurran** [comment]
@jcsteh this logic works okay for bold or italic, but what about font name, size, color etc... all can be changed within a cell according to @qchristensen. I think we'll need to consider supporting the concept of "mixed" for all formatting in NVDA if we want to fix this particular issue. It looks like most of the formatting like font name, font size etc are spoken raw in speech.py thus we could make them say "mixed" easily enough. However, code for bold and italic etc would need to change if we want it to say "mixed bold" for instance. Or, for bold and italic we could fallback to the original idea, though I feel that this is not consistent.

**Comment by jcsteh** [comment]
Good point. I think having support for mixed makes sense. I guess we could have a mixed constant in textInfos.

Note that when implementing, braille will need to be tweaked for this as well. We only currently set typeform for bold, italic, etc., but it almost certainly assumes that these are bool values, which won't be the case any more.

**Comment by Adriani90** [comment]
cc: @lukaszgo1, @leonardder
